### ISSUE TITLE
What are RIPs purpose and process?

### ISSUE THREAD

**Issue by slack-coder** [radicle-dev/rips#3]
What are RIPs for? They come across to me as design proposals for Radicle's Heartwood project. Take defining RAD_HOME for example. I recommend minimizing them to what's expected for interoperability for third party RAD applications.

RIPs appear to take from BIPs, and we missed an important aspect. BIPs have three purposes as shown below. Maybe we should adopt 'Process' as a use for RIPs? The [first BIP](https://github.com/bitcoin/bips/blob/master/bip-0001.mediawiki) is about introducing BIPs, defining process and what they are for.

  - Standards Track BIPs – Such types of BIPs entail making changes to the [network](https://en.bitcoinwiki.org/wiki/Network) protocol, [block](https://en.bitcoinwiki.org/wiki/Block), or [transaction](https://en.bitcoinwiki.org/wiki/Transaction) validation method. It also intends to affect the interoperability of the two versions of BIPs or [Bitcoin](https://en.bitcoinwiki.org/wiki/Bitcoin). This type of BIP certainly requires community consensus. An example of this is BIP 91.

  - Informational BIPs – Such types of BIPs highlight the design issues, general guidelines, and supporting information. Informational BIPs, as the name suggests, are just for information’s sake and can be taken seriously or ignored by the community. An example of this is BIP 32.

  - Process BIPs – These types of BIPs describe or propose a change in the process. They are similar to Standards Track BIPs and require community consensus. They can’t be ignored, but unlike Standards Track BIPs, they intend to be applied outside the Bitcoin protocol. An example of this is BIP 2.

**Comment by cloudhead** [comment]
> What are RIPs for? They come across to me as design proposals for Radicle's Heartwood project. Take defining RAD_HOME for example. I recommend minimizing them to what's expected for interoperability for third party RAD applications.

Spot on.

> RIPs appear to take from BIPs, and we missed an important aspect. BIPs have three purposes as shown below. Maybe we should adopt 'Process' as a use for RIPs? The [first BIP](https://github.com/bitcoin/bips/blob/master/bip-0001.mediawiki) is about introducing BIPs, defining process and what they are for.

That's right, though I disagree that any important aspect was missed. If we wish to categorize RIPs, we can do so, but for now they are all the equivalent of "Standards Track". The categorization is just a way of describing the scope of BIPs, and we can adopt a similar scope as BIPs if we find that useful in the future.

**Comment by FintanH** [comment]
> > What are RIPs for? They come across to me as design proposals for Radicle's Heartwood project. Take defining RAD_HOME for example. I recommend minimizing them to what's expected for interoperability for third party RAD applications.
> 
> Spot on.

In my mind, I would think `RAD_HOME` being standardised across implementations would make sense to know where to find the necessary, stored data. Are you saying that it doesn't make sense to standardise it?

**Comment by slack-coder** [comment]
> That's right, though I disagree that any important aspect was missed. If we wish to categorize RIPs, we can do so, but for now they are all the equivalent of "Standards Track". The categorization is just a way of describing the scope of BIPs, and we can adopt a similar scope as BIPs if we find that useful in the future.

Documenting RIP's scope somewhere will help newcomers and outsiders understand the context.  Its hard to intuit it specially since the naming is similar to Bitcoin Improvement Proposals.

**Comment by slack-coder** [comment]
> In my mind, I would think `RAD_HOME` being standardised across implementations would make sense to know where to find the necessary, stored data. Are you saying that it doesn't make sense to standardise it?

There was confusion about RIPs context for me.  So in the case of RIPs yes.  If we used the same scope of BIPs no, best leave it to the project maintainers as its a configuration parameter.

**Comment by cloudhead** [comment]
Yeah, RAD_HOME is a good example: I don't think it should be standardized in a RIP, since it's implementation dependent. For example, different radicle clients could use different configuration/env vars, but still all work together.

**Comment by cloudhead** [comment]
I'll write up a RIP#0 at some point with all the meta information.

**Comment by FintanH** [comment]
I guess I can see both ways making sense. The location of the storage is independent of the layout itself, on one hand. Being able to easily change clients without changing what RAD_HOME means, on the other. 

Thanks for the explanations :+1: 

I can update the storage RIP to not  mention RAD_HOME and look for similar culprits :)
